src/main/java/io/zeebe/monitor/repository/ProcessInstanceRepository.java

```diff
@@ -39,4 +48,19 @@ Page<ProcessInstanceEntity> findByParentProcessInstanceKey(
       long parentProcessInstanceKey, Pageable pageable);
 
   long countByParentProcessInstanceKey(long parentProcessInstanceKey);
+
+  @Query(value = "SELECT p.KEY_ FROM PROCESS_INSTANCE p" +
+          " WHERE p.STATE_ = :state" +
+          " AND p.END_ IS NOT NULL" +
+          " AND (CAST(:currentTimeMillis AS BIGINT) - p.END_) > :expirationIntervalMillis"
+          , nativeQuery = true)
+  List<Long> findAllExpiredIds(
```

Comment:
Bad naming. It's not clear from the method's name what is the role of the parameters (especially 'state')